beacon/v1/v1.go

```diff
@@ -0,0 +1,193 @@
+	"strconv"
+)
+
+var log = logrus.WithField("module", "v1")
+
+type V1HTTPClient struct {
+	api    *sling.Sling
+	client *http.Client
+}
+
+func (s *V1HTTPClient) GetVersion() (string, error) {
+	path := "v1/node/version"
+	type versionResponse struct {
+		Data struct {
+			Version string `json:"version,omitempty"`
+		} `json:"data,omitempty"`
+	}
+	response := new(versionResponse)
+	_, err := s.api.New().Get(path).ReceiveSuccess(response)
+	if err != nil {
+		return "", err
+	}
+	return response.Data.Version, nil
+}
+
+func (s *V1HTTPClient) GetGenesisTime() (int64, error) {
+	path := "v1/beacon/genesis"
+	type genesisResponse struct {
+		Data struct {
+			GenesisTime string `json:"genesis_time,omitempty"`
+		} `json:"data,omitempty"`
+	}
+	response := new(genesisResponse)
+	_, err := s.api.New().Get(path).ReceiveSuccess(response)
+	if err != nil {
+		return 0, err
+	}
+	genesisTime, err := strconv.ParseInt(response.Data.GenesisTime, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return genesisTime, nil
+}
+
+func (s *V1HTTPClient) GetPeerCount() (int64, error) {
+	path := "v1/node/peers"
+	type peersResponse struct {
+		Data []struct {
+		} `json:"data,omitempty"`
+	}
+	response := new(peersResponse)
+	_, err := s.api.New().Get(path).ReceiveSuccess(response)
+	if err != nil {
+		return 0, err
+	}
+	return int64(len(response.Data)), nil
+}
+
+func (s *V1HTTPClient) GetAttestationsInPoolCount() (int64, error) {
+	path := "v1/beacon/pool/attestations"
+	type attestationsResponse struct {
+		Data []struct {
+		} `json:"data,omitempty"`
+	}
+	response := new(attestationsResponse)
+	_, err := s.api.New().Get(path).ReceiveSuccess(response)
+	if err != nil {
+		return 0, err
+	}
+	return int64(len(response.Data)), nil
+}
+
+func (s *V1HTTPClient) GetSyncStatus() (bool, error) {
+	path := "v1/node/syncing"
+	type syncingResponse struct {
+		Data struct {
+			SyncDistance string `json:"sync_distance,omitempty"`
+		} `json:"data,omitempty"`
+	}
+	response := new(syncingResponse)
+	_, err := s.api.New().Get(path).ReceiveSuccess(response)
+	if err != nil {
+		return false, err
+	}
+	syncDistance, err := strconv.ParseUint(response.Data.SyncDistance, 10, 64)
+	if err != nil {
+		return false, err
+	}
+	return syncDistance != 0, nil
+}
+
+func (s *V1HTTPClient) GetChainHead() (*types.ChainHead, error) {
+
+	typesChainHead := new(types.ChainHead)
+
+	headRootPath := "v1/beacon/blocks/head/root"
+	type headRootType struct {
+		Data struct {
+			HeadBlockRoot string `json:"root,omitempty"`
+		} `json:"data,omitempty"`
+	}
+	headRootResponse := new(headRootType)
+	_, err := s.api.New().Get(headRootPath).ReceiveSuccess(headRootResponse)
+	if err != nil {
+		return nil, err
+	}
+	typesChainHead.HeadBlockRoot = headRootResponse.Data.HeadBlockRoot
+
+	slot, err := s.getBlockSlot(typesChainHead.HeadBlockRoot)
+	if err != nil {
+		return nil, err
+	}
+	typesChainHead.HeadSlot = slot
+
+	finalityCheckpointsPath := "v1/beacon/state/head/finality_checkpoints"
```

Comment:
If this request has no arguments/params, then it may not match the head-root. E.g. it reports head-root of 1 chain, and justification of another. I would prefer it if we can improve the v1 API to merge these common data points into a single request/response. How about adding `block root` and `slot` to the finality endpoints response?

Reply:
It would quickly update in that case, but yeah, I think we should add block root to this response (so we have reference for which head is this response), not sure if we should put slot as well. Will ask in WG on telegram

Reply:
amazing work guys, thanks
should i hold the merge until you decide what you decide what to do about these endpoints?

a simple endpoint for chain head like in the current temporary implementations would also be good to have, this is the endpoint i end up polling most in all projects

Reply:
yeah, let's hold on for a bit, till we figure out the head endpoint. Other clients are still lagging with standardized api impl so we have time.

I would also like to replace polling with event stream:
https://docs.google.com/spreadsheets/d/1kVIx6GvzVLwNYbcd-Fj8YUlPf4qGrWUlS35uaTnIAVg/edit#gid=215290094&range=9:9

But I need to implement it in lodestar first :smile:

Reply:
i would love to see event stream widely implemented, i hate polling :P

i've given @protolambda write access to the repo just in case i'm unresponsive

Reply:
Maybe we should just merge this in the meantime, to make sure lodestar can join Medalla testnet with eth2stats. The API can improve independent of this change, and I'm available for more PR reviews if necessary.